### DISCUSSION TITLE
How do i add an `n` pixels gap between two widgets?

### DISCUSSION THREAD

**Discussion by abhijeetbhagat** [emilk/egui#269] [Q&A] [answered]
I've tried `ui.spacing_mut().item_spacing.y = 50.0;` between a button and a textbox above it, but it's not working.

**Comment by emilk** [comment, marked as answer]
`item_spacing` should work if you call it _before_ adding the _first_ of the two widgets (that's a bit counter-intuitive and is because spacing is applied _after_ adding a widget)

But a more straightforward way is to call `ui.advance_cursor(50.0);`. I will rename that to `ui.add_space(50.0);` in the next release of egui.

**Reply by abhijeetbhagat** [reply]
awesome! thanks!
